src/test/java/tests/ProductsTest.java

```diff
@@ -6,43 +6,50 @@
 import pages.Constants.LoginPageConstants;
 import tests.Constants.ProductsTestConstants;
 
-import java.lang.reflect.Array;
 import java.util.*;
 
 public class ProductsTest extends BaseTest {
+    public static final String ADD_PRODUCT_CLASS = "btn_primary";
+    public static final String REMOVE_PRODUCT_CLASS = "btn_secondary";
+    public static final String PRODUCT_TYPE_SORT = "az";
```

Comment:
It's not correct to place constants in Test classes. Here should be only Test methods.

I suggest you not to put html attributes  "btn_primary", "btn_secondary", "az" in constants.

It's better to refactor this method:
    
public void clickProductItem(String productName, String nameClick) {
        getProduct(productName)
                .findElement(By.xpath(String.format(PRODUCT_TO_CART_BTN, nameClick))).click();
    }

**Option 1:** Divide it into 2 separate methods

public void clickAddProductItem(String productName) {
        getProduct(productName).findElement(By.xpath(PRODUCT_ADD_TO_CART_BTN)).click();
    }

public void clickRemoveProductItem(String productName) {
        getProduct(productName).findElement(By.xpath(PRODUCT_REMOVE_FROM_CART_BTN)).click();
    }

    public static final String PRODUCT_ADD_TO_CART_BTN= "//button[contains(@class, 'btn_primary')]";
    public static final String PRODUCT_REMOVE_FROM_CART_BTN= "//button[contains(@class, 'btn_secondary')]";

**Option 2:** Pass boolean value
    public void clickAddOrRemoveProductItem(String productName, boolean isAddToCart) {
        if(isAddToCart) {
            getProduct(productName).findElement(By.xpath(PRODUCT_ADD_TO_CART_BTN)).click();
        } else {
            getProduct(productName).findElement(By.xpath(PRODUCT_REMOVE_FROM_CART_BTN)).click();
        }
    }

Reply:
done